server/app/views/admin/migration/AdminImportViewPartial.java

```diff
@@ -124,14 +141,19 @@ public DomContent renderProgramData(
   }
 
   /** Renders a message saying the program was successfully saved. */
-  public DomContent renderProgramSaved(String programName, Long programId) {
+  public DomContent renderProgramSaved(Http.Request request, String programName, Long programId) {
+    String successText =
+        programName
+            + " and its questions have been imported to your program dashboard. To view it,  visit"
+            + " the program dashboard.";
+    if (settingsManifest.getNoDuplicateQuestionsForMigrationEnabled(request)) {
+      successText += " Before you import another program, you will need to publish all drafts.";
```

Comment:
praise: clever way to handle this!

Reply:
thought/question (non-blocking): It's still possible for admins to re-enter the import flow and try to import a program that already contains draft questions. in that case they'll see this error. do you think we should handle this and give them a more helpful error message or is this enough? 
<img width="869" alt="image" src="https://github.com/user-attachments/assets/fbe42f8b-37b7-4b4f-ba2a-283d0668c6ea">

Reply:
Yeah, good point! Updated to catch that case and give a helpful error.